database-commons/src/main/java/io/cdap/plugin/util/DBUtils.java

```diff
@@ -163,17 +163,23 @@ public static void deregisterAllDrivers(Class<? extends Driver> driverClass)
 
   public static void validateJDBCPluginPipeline(PipelineConfigurer pipelineConfigurer, ConnectionConfig config,
                                                 String jdbcPluginId) {
-    if (Objects.isNull(config.user) && Objects.nonNull(config.password)) {
-      throw new InvalidConfigPropertyException("user", "user is null. Please provide both user name and password if " +
-        "database requires authentication. If not, please remove password and retry.");
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    if (!config.containsMacro(DBConfig.USER) && !config.containsMacro(DBConfig.PASSWORD) &&
+      Objects.isNull(config.user) && Objects.nonNull(config.password)) {
+      collector.addFailure("Username and password should be provided together.", "Please provide both " +
+        "user name and password, if database supports it or remove password.")
+        .withConfigProperty(ConnectionConfig.USER).withConfigProperty(ConnectionConfig.PASSWORD);
```

Comment:
this should only highlight the user property, since there is nothing wrong with providing a password. The message can also be simplified to just say that a user name is required when a password is given.

Reply:
fixed